test/sqllogictest/not-null-propagation.slt

```diff
@@ -321,19 +321,30 @@ Explained Query:
     Project (#0, #1, #8, #9, #7) // { types: "(integer?, integer?, numeric?, numeric?, integer list?)" }
       Map ((bigint_to_numeric(#2) / bigint_to_numeric(case when (#3 = 0) then null else #3 end)), sqrtnumeric(case when ((#4) IS NULL OR (#5) IS NULL OR (case when (#6 = 0) then null else #6 end) IS NULL OR (case when (0 = (#6 - 1)) then null else (#6 - 1) end) IS NULL) then null else greatest(((#4 - ((#5 * #5) / bigint_to_numeric(case when (#6 = 0) then null else #6 end))) / bigint_to_numeric(case when (0 = (#6 - 1)) then null else (#6 - 1) end)), 0) end)) // { types: "(integer?, integer?, bigint?, bigint, numeric?, numeric?, bigint, integer list?, numeric?, numeric?)" }
         Union // { types: "(integer?, integer?, bigint?, bigint, numeric?, numeric?, bigint, integer list?)" }
-          Get l0 // { types: "(integer, integer, bigint, bigint, numeric, numeric, bigint, integer list)" }
+          Project (#5, #6, #0..=#4, #7) // { types: "(integer, integer, bigint, bigint, numeric, numeric, bigint, integer list)" }
+            Get l1 // { types: "(bigint, bigint, numeric, numeric, bigint, integer, integer, integer list)" }
           Map (null, null, null, 0, null, null, 0, null) // { types: "(integer?, integer?, bigint?, bigint, numeric?, numeric?, bigint, integer list?)" }
             Union // { types: "()" }
               Negate // { types: "()" }
                 Project () // { types: "()" }
-                  Get l0 // { types: "(integer, integer, bigint, bigint, numeric, numeric, bigint, integer list)" }
+                  Get l1 // { types: "(bigint, bigint, numeric, numeric, bigint, integer, integer, integer list)" }
```

Comment:
This looks like a projection pushdown failure. Or .. given that it only happens later in the pipeline, and apparent regression that gets fixed by projection pushdown potentially. The new `Project (#5 ... ` stuff is just re-ordering the aggregate fields from the `l1 = CrossJoin .. ` term (and potentially projecting away duplicate keys, but it is a cross join so .. no keys).

Reply:
`ProjectionPushdown` is able to push reorderings through some operators, but not through others. Specifically, it doesn't do it for `Get`s, because `gets` keeps a `BTreeSet` of columns for each id, rather than some other data structure that would be able to represent a reordering.

If it's desirable to push down reorderings, I can rework how `ProjectionPushdown` behaves with respect to reordering as part of fixing https://github.com/MaterializeInc/materialize/issues/17522 (which will involve a pretty big refactor of `ProjectionPushdown` anyway). But my feeling is that it's generally pretty hard to tell where a reordering should optimally be, because this is somewhat influenced by which operator is able to absorb it later during lowering.